### PR TITLE
deprecate: Deprecate ip_addresses attribute in project resource and data source

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

Deprecates ip_addresses in resource and datasource mongodbatlas_project

**Agustin Edit:**
Verified that with the new plan modifier configured in `ip_addresses` the plan no longer show details associated to this attribute. This will avoid further client confusion until the attribute is removed completely.
Screenshot 2024-09-03 at 11 25 57

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [X] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

**Review by EspenAlbert — APPROVED** [review]
LGTM

**Review by lantoli — APPROVED** [review]

File: .changelog/2541.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+```release-note:note
+resource/mongodbatlas_project: Deprecates the `ip_addresses` attribute. The new `mongodbatlas_project_ip_addresses` data source is avaible to obtain this information.
```
Comment (davidhou17) [inline .changelog/2541.txt]:
```suggestion
resource/mongodbatlas_project: Deprecates the `ip_addresses` attribute. Use the new `mongodbatlas_project_ip_addresses` data source to obtain this information instead.
```

File: docs/data-sources/project.md [hunk 2]
```diff
@@ -81,7 +81,7 @@ In addition to all arguments above, the following attributes are exported:
 * `tags` - Map that contains key-value pairs between 1 to 255 characters in length for tagging and categorizing the project. To learn more, see [Resource Tags](https://www.mongodb.com/docs/atlas/tags/)
 * `teams` - Returns all teams to which the authenticated user has access in the project. See [Teams](#teams).
 * `limits` - The limits for the specified project. See [Limits](#limits).
-* `ip_addresses` - IP addresses in a project categorized by services. See [IP Addresses](#ip-addresses).
+* `ip_addresses` - IP addresses in a project categorized by services. See [IP Addresses](#ip-addresses). **WARNING:** this attribute is deprecated and will be removed in version 1.21.0. Please transition to using `mongodbatlas_project_ip_addresses` data source.
```
Comment (davidhou17) [inline docs/data-sources/project.md]:
```suggestion
* `ip_addresses` - IP addresses in a project categorized by services. See [IP Addresses](#ip-addresses). **WARNING:** this attribute is deprecated and will be removed in version 1.21.0. Use the `mongodbatlas_project_ip_addresses` data source instead.
```

**Review by davidhou17 — COMMENTED** [review]
Can we apply the following clarifications across these descriptions?

File: .changelog/2541.txt (hunk 1 above)
Comment (AgustinBettati) [reply .changelog/2541.txt]: thanks! applied for all cases

File: docs/data-sources/project.md (hunk 2 above)
Comment (AgustinBettati) [reply docs/data-sources/project.md]: same for this suggestion.

**Review by davidhou17 — APPROVED** [review]

**Pull request merged**